src/shared/redux/modules/member.js

```diff
@@ -37,18 +40,30 @@ const reducer: Reducer<State, Action> =
       case DEACTIVATED_MEMBER:
         return (
           { ...member
-          , activation_status: { value: 'deactivated', initial_value: member.activation_status.initial_value }
+          , activation_status: { value: 'deactivated', initialValue: member.activation_status.initialValue }
           , deletion_date: { value: new Date().toISOString() }
           }
         )
       case REACTIVATED_MEMBER:
         return (
           { ...member
-          , activation_status: { value: 'activated', initial_value: member.activation_status.initial_value }
+          , activation_status: { value: 'activated', initialValue: member.activation_status.initialValue }
           , deletion_date: {}
           , deletion_reason: { value: null }
           }
         )
+      case UPDATED_MEMBERSHIP_TYPE:
```

Comment:
I'm not sure what you mean by different kinds of updates. Both `date_membership_type_changed` and `life_payment_date` are triggered by changing the `membership_type`. Surely if I had two cases, they would both get the same payload?